Auth pre-flight in claude-wrapper.sh is a single point of failure for shepherd

## Problem

When `claude auth status --json` hangs during the `claude-wrapper.sh` pre-flight check, **all** shepherd-spawned sessions fail — even though the Claude API endpoint is reachable and the CLI itself may work fine.

This was observed during shepherd run for #2396: the wrapper's auth check timed out 3 times (15s timeout × 3 attempts), causing the wrapper to exit before writing the `# CLAUDE_CLI_START` sentinel. The shepherd then retried the full session 3 more times with backoff, but every retry hit the same auth timeout. Total wasted time: ~5 minutes per phase (curator + builder).

## Observed Behavior

From builder log:
```
[INFO] Claude CLI found: /opt/homebrew/bin/claude
[INFO] API endpoint reachable (curl)
[INFO] Authentication check timed out (attempt 1/3), retrying in 2s...
[INFO] Authentication check timed out (attempt 2/3), retrying in 5s...
[ERROR] Authentication check timed out after 3 attempts
[ERROR] Authentication pre-flight check failed
```

The API endpoint was reachable (curl check passed), but `claude auth status --json` hung. The shepherd's outer retry loop (3 retries with 2/4/8s backoff) retried the same failing command, wasting another ~5 minutes before giving up.

## Root Cause

`claude-wrapper.sh` runs `timeout 15 bash -c 'CLAUDECODE="" claude auth status --json 2>&1'` as a mandatory pre-flight gate. When this command hangs (which can happen due to CLI internal state, token refresh issues, or system load), no session can start — the wrapper exits before invoking Claude CLI.

## Suggested Fix

Consider one or more of:

1. **Skip auth check in `--dangerously-skip-permissions` mode** — the session already has auth credentials; let the CLI itself fail-fast if not authenticated rather than gating on a separate check
2. **Make auth check optional / best-effort** — log a warning but proceed to CLI invocation, letting the CLI handle auth errors
3. **Accept cached auth results more aggressively** — the wrapper already has a cache mechanism; extend the cache TTL or trust it when the API endpoint is reachable

## Files

- `.loom/scripts/claude-wrapper.sh` (`check_auth_status()`, auth cache logic)
- `loom-tools/src/loom_tools/shepherd/base.py` (instant-exit detection, retry loop)

--- a/defaults/scripts/claude-wrapper.sh
+++ b/defaults/scripts/claude-wrapper.sh
@@ -59,6 +59,9 @@ TERMINAL_ID="${LOOM_TERMINAL_ID:-}"
 # Note: WORKSPACE may fail if CWD is invalid at startup - recover_cwd handles this
 WORKSPACE="${LOOM_WORKSPACE:-$(pwd 2>/dev/null || echo "$HOME")}"
 
+# Whether --dangerously-skip-permissions was passed (detected in main())
+SKIP_PERMISSIONS_MODE=false
+
 # Retry state file for external observability (see issue #2296).
 # When TERMINAL_ID is set, the wrapper writes its retry/backoff state to this
 # file so agent-wait-bg.sh and the shepherd can distinguish "wrapper retrying"
@@ -956,11 +959,15 @@ run_preflight_checks() {
     check_api_reachable  # Non-fatal, just logs
 
     if ! check_auth_status; then
-        log_error "Authentication pre-flight check failed"
-        # Write sentinel so the shepherd can distinguish auth failures from
-        # generic instant-exits and avoid futile retries.  See issue #2508.
-        echo "# AUTH_PREFLIGHT_FAILED" >> "${LOG_FILE}"
-        return 1
+        if [[ "${SKIP_PERMISSIONS_MODE}" == "true" ]]; then
+            log_warn "Authentication pre-flight check failed (non-fatal in --dangerously-skip-permissions mode)"
+        else
+            log_error "Authentication pre-flight check failed"
+            # Write sentinel so the shepherd can distinguish auth failures from
+            # generic instant-exits and avoid futile retries.  See issue #2508.
+            echo "# AUTH_PREFLIGHT_FAILED" >&2
+            return 1
+        fi
     fi
 
     if ! check_mcp_server; then
@@ -981,6 +988,14 @@ main() {
     log_info "Arguments: $*"
     log_info "Workspace: ${WORKSPACE}"
     [[ -n "${TERMINAL_ID}" ]] && log_info "Terminal ID: ${TERMINAL_ID}"
+
+    # Detect --dangerously-skip-permissions flag (automated agent mode)
+    for arg in "$@"; do
+        if [[ "$arg" == "--dangerously-skip-permissions" ]]; then
+            SKIP_PERMISSIONS_MODE=true
+            break
+        fi
+    done
 
     # Run pre-flight checks
     if ! run_preflight_checks; then

